src/algorithms/calorimetry/ImagingTopoClusterConfig.h

```diff
@@ -29,4 +34,18 @@ namespace eicrecon {
 
   };
 
+  std::istream& operator>>(std::istream& in, ImagingTopoClusterConfig::ELayerMode& layerMode) {
+    std::string s;
+    in >> s;
+
+    if (s == "etaphi" or s == "0") {
```

Comment:
If we have users specify strings on command line, then I don't think it makes sense to allow "0" and "1". Those are not descriptive in the face of a better, existing alternative.

Reply:
When I set the values of layerMode using an enum value (as shown below in src/detectors/ZDC/ZDC.cc), it converts it to an int and then to a string representation of that int at runtime.  I don't know how why it's doing this.  

app->Add(new JOmniFactoryGeneratorT<ImagingTopoCluster_factory>(
            "HcalFarForwardZDCImagingProtoClusters", {"HcalFarForwardZDCSubcellHits"}, {"HcalFarForwardZDCImagingProtoClusters"},
            {
                .neighbourLayersRange = 1,
                .localDistXY = {0.76*side_length, 0.76*side_length*sin(M_PI/3)},
                .layerDistXY = {0.76*side_length, 0.76*side_length*sin(M_PI/3)},
                .layerMode=eicrecon::ImagingTopoClusterConfig::ELayerMode::xy,
                .sectorDist = 10.0 * dd4hep::cm,
                .minClusterHitEdep = 100.0 * dd4hep::keV,
                .minClusterCenterEdep = 3.0 * dd4hep::MeV,
                .minClusterEdep = 11.0 * dd4hep::MeV,
                .minClusterNhits = 100,
            },

Reply:
Right, now I recall. JANA2 does a roundtrip through string representation land, so we don't just need the istream but also an ostream operator for the enum. If we have an overload it won't fall back to the representing type of int. I'm on my phone and can't easily write a code snippet, but I assume you get the idea.

Reply:
Ok, I'll look up how to make a ostream representation and will try to implement that

Reply:
tried using a ostream representation, but it was still giving me "1" or "0" for the layer mode at runtime.